feat(content-docs): add custom props front matter (#6619)
facebook/docusaurus: website/docs/guides/docs/sidebar/autogenerated.md

## Autogenerated sidebar metadata {#autogenerated-sidebar-metadata}

For handwritten sidebar definitions, you would provide metadata to sidebar items through `sidebars.js`; for autogenerated, Docusaurus would read them from the item's respective file. In addition, you may want to adjust the relative position of each item because, by default, items within a sidebar slice will be generated in **alphabetical order** (using file and folder names).

> **For docs**: use additional front matter. The `position`, `label`, and `className` attributes are declared using `sidebar_position`, `sidebar_label`, and `sidebar_class_name` front matter, respectively. (Custom props can also be supplied via `sidebar_custom_props` front matter.)

Reviewer: ```suggestion
**For docs**: use additional front matter. The `label`, `className`, and `customProps` attributes are declared using `sidebar_label`, `sidebar_class_name`, and `sidebar_custom_props` front matter, respectively. There's also an additional `sidebar_position` attribute.
```

We are actually comparing these attributes to how they are declared in `sidebars.js`. In the latter case the `position` attribute doesn't exist, so we mention it separately.

Author: Ah, I see.  I just thought the language was super awkward here, but will fix.

Reviewer: I read the thread in #6590 and the diff in this PR only to get myself even more confused 😅 `sidebar_custom_props` is effective even when the sidebar is not autogenerated, similar to `sidebar_label`. In this case I would mention it in "sidebar items" section as well, and also point out that this can be used to pass custom doc metadata to the client side (as is the original use-case)

Author: Updated, let me know if this is better!

Author: > I read the thread in #6590 and the diff in this PR only to get myself even more confused 😅 `sidebar_custom_props` is effective even when the sidebar is not autogenerated, similar to `sidebar_label`. In this case I would mention it in "sidebar items" section as well, and also point out that this can be used to pass custom doc metadata to the client side (as is the original use-case)

Oh didn't see this reply until just now.  And yes, that sounds reasonable.  Will update again tomorrow, it's getting late here. :)

Reviewer: And also, my goal is to not make `sidebar_custom_props` appear different from `sidebar_label` in the documentation, so they should always be listed in parallel.

Author: > And also, my goal is to not make `sidebar_custom_props` appear different from `sidebar_label` in the documentation, so they should always be listed in parallel.

Yup, I realized that from your previous comment, there was just some delay in it appearing and I didn't see it until after my last commit.  Will update again shortly.

Author: Made changes as requested, please take a look and let me know if anything else needs to be fixed!
